### ISSUE TITLE
Setup tries to install non-existing version of Gradle

### ISSUE THREAD

**Issue by pparkkin** [openwhisk/openwhisk#767]
> note: obfuscate all passwords, credentials or other confidential information

## Environment details:

Native Ubuntu. Ubuntu 16.04.

    $ docker --version
    Docker version 1.9.1, build a34a1d5
    $ vagrant --version
    The program 'vagrant' is currently not installed.

## Steps to reproduce the issue:

    # Install git if it is not installed
    apt-get install git
    
    # Clone openwhisk
    git clone https://github.com/openwhisk/openwhisk.git
    
    # Change current directory to openwhisk
    cd openwhisk
    
    # Install all required software
    (cd tools/ubuntu-setup && source all.sh)

## Provide the expected results and outputs:

I would expect the output to not be an error.

## Provide the actual results and outputs:

Bunch of other output, and this.

```
Reading package lists... Done
Building dependency tree
Reading state information... Done
E: Unable to locate package gradle-2.7
E: Couldn't find any package by glob 'gradle-2.7'
E: Couldn't find any package by regex 'gradle-2.7'
```

**Comment by psuter** [comment]
The problem seems to be that `gradle-2.7` is not available for Ubuntu 16.04. Alternative method to uncover the issue (e.g. for Mac users); attempt to build the following docker image, mimicking the steps in `gradle.sh`:

    FROM ubuntu:16.04
    ENV DEBIAN_FRONTEND noninteractive
    RUN apt-get -y update
    RUN apt-get install -y software-properties-common
    RUN apt-get install -y python-software-properties
    RUN add-apt-repository -y ppa:cwchien/gradle
    RUN apt-get -y update
    RUN apt-cache search gradle
    RUN apt-get install -y gradle-2.7

The `apt-cache` command shows that the available versions are:

    gradle - Gradle is a Groovy based build system
    gradle-ppa - Gradle is a Groovy based build system
    gradle-2.9 - Gradle is a Groovy based build system
    gradle-2.10 - Gradle is a Groovy based build system
    gradle-2.11 - Gradle is a Groovy based build system
    gradle-2.12 - Gradle is a Groovy based build system
    gradle-2.13 - Gradle is a Groovy based build system
    gradle-2.14 - Gradle is a Groovy based build system

...and the `apt-get install` command fails as a consequence.

@pparkkin you can for now almost certainly work around the issue by modifying `tools/ubuntu/gradle.sh` to install `gradle-2.9`, for instance.

**Comment by pparkkin** [comment]
Great. Thanks!

**Comment by markusthoemmes** [comment]
@psuter: We are in fact bundling a `gradlew` launcher. We use Gradle 2.12 for that, so we should update `ubuntu-setup.sh`

**Comment by domdom82** [comment]
Could we instead drop installing gradle altogether and only use `gradlew`? One less dep to manage...

**Comment by rabbah** [comment]
No longer valid - fixed by #961.
